p2p/src/connection/outgoing/p2p_connection_outgoing_reducer.rs

```diff
@@ -52,8 +58,13 @@ impl P2pConnectionOutgoingState {
                                 &opts,
                             )),
                             identify: None,
+                            on_connect_success: None,
                         });
 
+                if let Some(on_connect_success) = on_success {
+                    peer_state.on_connect_success = Some(on_connect_success);
```

Comment:
To make sure I understand here, this is required because the peer state may exists because this is an incoming connection, so in such case there is no need for the callback?

Reply:
Yes